apps/server/src/modules/h5p-editor/uc/h5p.uc.ts

```diff
@@ -0,0 +1,404 @@
+import { ICurrentUser } from '@src/modules/authentication';
+import { AuthorizationContext, AuthorizationContextBuilder } from '@src/modules/authorization';
+import { AuthorizationReferenceService } from '@src/modules/authorization/domain';
+import { UserService } from '@src/modules/user';
+import { Request } from 'express';
+import { Readable } from 'stream';
+import { AjaxGetQueryParams, AjaxPostBodyParams, AjaxPostQueryParams } from '../controller/dto';
+import { H5PContentParentType } from '../entity';
+import { H5PContentMapper } from '../mapper/h5p-content.mapper';
+import { H5PContentRepo } from '../repo';
+import { LibraryStorage } from '../service';
+import { LumiUserWithContentData } from '../types/lumi-types';
+import { H5PErrorMapper } from '../mapper/h5p-error.mapper';
+
+@Injectable()
+export class H5PEditorUc {
+	constructor(
+		private readonly h5pEditor: H5PEditor,
+		private readonly h5pPlayer: H5PPlayer,
+		private readonly h5pAjaxEndpoint: H5PAjaxEndpoint,
+		private readonly libraryService: LibraryStorage,
+		private readonly userService: UserService,
+		private readonly authorizationReferenceService: AuthorizationReferenceService,
+		private readonly h5pContentRepo: H5PContentRepo
+	) {}
+
+	private async checkContentPermission(
+		userId: EntityId,
+		parentType: H5PContentParentType,
+		parentId: EntityId,
+		context: AuthorizationContext
+	) {
+		const allowedType = H5PContentMapper.mapToAllowedAuthorizationEntityType(parentType);
+		await this.authorizationReferenceService.checkPermissionByReferences(userId, allowedType, parentId, context);
+	}
+
+	private fakeUndefinedAsString = () => {
+		const value = undefined as unknown as string;
+		return value;
+	};
+
+	/**
+	 * Returns a callback that parses the request range.
+	 */
+	private getRange(req: Request) {
+		return (filesize: number) => {
+			const range = req.range(filesize);
+
+			if (range) {
+				if (range === -2) {
+					throw new BadRequestException('invalid range');
+				}
+
+				if (range === -1) {
+					throw new BadRequestException('unsatisfiable range');
+				}
+
+				if (range.length > 1) {
+					throw new BadRequestException('multipart ranges are unsupported');
+				}
+
+				return range[0];
+			}
+
+			return undefined;
+		};
+	}
+
+	// eslint-disable-next-line consistent-return
+	public async getAjax(query: AjaxGetQueryParams, currentUser: ICurrentUser) {
+		const user = this.changeUserType(currentUser);
+		const language = await this.getUserLanguage(currentUser);
+		const h5pErrorMapper = new H5PErrorMapper();
+
+		try {
+			const result = await this.h5pAjaxEndpoint.getAjax(
+				query.action,
+				query.machineName,
+				query.majorVersion,
+				query.minorVersion,
+				language,
+				user
+			);
+			return result;
+		} catch (err) {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+			h5pErrorMapper.mapH5pError(err);
+		}
+	}
+
+	// eslint-disable-next-line consistent-return
+	public async postAjax(
```

Comment:
quite a few functions here that do not define their return type. Please define them, so the api is clear, at the very least for all public functions.

also please fix the inconsistent return and remove the eslint comment.

Reply:
resolved